spyder/plugins/variableexplorer/plugin.py

```diff
@@ -47,7 +47,19 @@ def get_icon(self):
         return self.create_icon('dictedit')
 
     def on_initialize(self):
-        pass
+        widget = self.get_widget()
+        widget.sig_open_preferences_requested.connect(
+            self._open_preferences
+        )
+
+    def _open_preferences(self):
+        """Open the Preferences dialog in the Variable Explorer section."""
+        self._main.show_preferences()
+        preferences = self._main.preferences
+        container = preferences.get_container()
+        dlg = container.dialog
+        index = dlg.get_index_by_name(self.NAME)
+        dlg.set_current_index(index)
```

Comment:
```suggestion
        self._main.show_preferences()
        preferences = self.get_plugin(Plugins.Preferences)
        if preferences:
            container = preferences.get_container()
            dlg = container.dialog
            index = dlg.get_index_by_name(self.NAME)
            dlg.set_current_index(index)
```

* Use the SpyderPluginV2 API to get Preferences, which is preferred to grab it directly from the main window.
* Check if it's available before using it.

Reply:
Just in case, I think this was done following the logic at https://github.com/spyder-ide/spyder/blob/4d51bddbf225a148d5b2796cef938cf69f0cf527/spyder/plugins/maininterpreter/plugin.py#L112-L119

So maybe a change there to use `get_plugin` there too for consistency could be worthy

Reply:
Yep, I saw that. If you want to do these changes there, you're welcome too (I probably implemented that before the Preferences plugin was available).